SubEthaEdit-Mac/Source/PlainTextDocument.m

```diff
@@ -3787,13 +3844,50 @@ - (NSData *)dataOfType:(NSString *)aType error:(NSError **)outError {
 	return data;
 }
 
+- (BOOL)onDiskRepresentationHasChanged {
+    NSURL *fileURL = self.fileURL;
+    NSString *fileName = [fileURL path];
+    
+    NSDictionary *fattrs = [[NSFileManager defaultManager] attributesOfItemAtPath:fileName error:nil];
+    
+    if (!fattrs) {
+        return YES;
+    }
+    
+    // If the modificationDate has not changed we can assume, that the content is still the same
+    if ([[fattrs fileModificationDate] compare:[[self fileAttributes] fileModificationDate]] == NSOrderedSame) {
+        return NO;
+    }
+    
+    // If the file size is different, the contents also differ for sure
+    if (fattrs.fileSize != self.fileAttributes.fileSize) {
+        return YES;
+    }
+    
+    NSData *fileData;
+    if(I_flags.isSEEText) {
+        NSMutableData *bundleData = [NSMutableData new];
+        NSData *textData = [NSData dataWithContentsOfURL:[fileURL URLByAppendingPathComponent:@"plain.txt"]];
+        NSData *metaData = [NSData dataWithContentsOfURL:[fileURL URLByAppendingPathComponent:@"collaborationdata.bencoded"]];
+        if (textData && metaData) {
+            [bundleData appendData:metaData];
+            [bundleData appendData:textData];
+            fileData = bundleData;
+        }
+    } else {
+        fileData = [NSData dataWithContentsOfURL:self.fileURL];
+    }
+    
+    return ![[fileData md5Data] isEqualToData:I_fileHash];
+    
+}
+
 - (BOOL)TCM_validateDocument {
     NSString *fileName = [[self fileURL] path];
     DEBUGLOG(@"FileIOLogDomain", SimpleLogLevel, @"Validate document: %@", fileName);
 
-    NSDictionary *fattrs = [[NSFileManager defaultManager] attributesOfItemAtPath:fileName error:nil];
-
-    if ([[fattrs fileModificationDate] compare:[[self fileAttributes] fileModificationDate]] == NSOrderedSame) {
+    
+    if (![self onDiskRepresentationHasChanged] || ![NSFileManager.defaultManager fileExistsAtPath:fileName]) {
```

Comment:
The order of the two items in the if probably should change if we treat a non-existing file as no change anyways.